pdns/dnsdist-tcp.cc

```diff
@@ -750,6 +754,13 @@ static void handleQuery(std::shared_ptr<IncomingTCPConnectionState>& state, cons
     TCPResponse response;
     response.d_selfGenerated = true;
     response.d_buffer = std::move(state->d_buffer);
+    setIDStateFromDNSQuestion(response.d_idstate, dq, std::move(qname));
+    response.d_idstate.origID = dh->id;
+    response.d_idstate.cs = state->d_ci.cs;
+
+    DNSResponse dr = makeDNSResponseFromIDState(response.d_idstate, response.d_buffer);
```

Comment:
That sound a bit wasteful, don't we already have a pointer to the header above (before the moves)?